src/frontend/fir.cpp

```diff
@@ -1131,6 +1146,24 @@ namespace graphit {
             return node;
         }
 
+        void ConstantVectorExpr::copy(FIRNode::Ptr node) {
+            const auto constantVectorExpr = to<ConstantVectorExpr>(node);
+            Expr::copy(constantVectorExpr);
+            numElements = constantVectorExpr->numElements;
+
+            for (const auto &el : constantVectorExpr->vectorElements) {
+                vectorElements.push_back(el ? el->clone<Expr>() : Expr::Ptr());
```

Comment:
Can we clean this up for better readability? Also use `nullptr` instead of `Expr::Ptr()`. My guess is this might cause memory issues. 

On second thought, why do we even need this check? Isn't el guaranteed to be non-null? What does the default constructor for a shared_ptr return?